reflect.go

```diff
@@ -328,7 +328,7 @@ func (r *Reflector) reflectStruct(definitions Definitions, t reflect.Type) *Type
 	if r.AllowAdditionalProperties {
 		st.AdditionalProperties = []byte("true")
 	}
-	definitions[t.Name()] = st
+	definitions[t.String()] = st
```

Comment:
You shouldn't use String(), because it's basically intended for humans - it gets shortened, it's not guaranteed to be unique, etc. Please try PkgPath().